grid-proxy/tests/queries/local_client.go

```diff
@@ -398,6 +398,41 @@ func (g *GridProxyClientimpl) Counters(filter proxytypes.StatsFilter) (res proxy
 	return
 }
 
+func gpuSatisfies(data *DBData, node node, f proxytypes.NodeFilter) bool {
+	gpu := data.gpus[node.twin_id]
+
+	if f.GpuDeviceName != nil {
+		if !strings.Contains(strings.ToLower(gpu.device), *f.GpuDeviceName) {
```

Comment:
I think `contains` here is not the best way for GPU device name as some GPU names get a suffix `ti`, `super` or `xtx` so there might be devices that get reported inaccurately.

for example: 4090 and 4090ti, 7900xt and 7900xtx

Reply:
This is only done for a specific test case with known input, the tests are in the middle of a refactor right now so this isn't a problem.